### PR TITLE
Updating Go version and modify tfproviderlint actions to use specified go version

### PR DESCRIPTION
**Pull request by eduardodfmex**
This pull request primarily focuses on updating versions and commenting out a section of the workflow. The changes include updating the base image and Go version in the `.devcontainer/devcontainer.json` and `go.mod` files respectively, and commenting out the `linter` job in the `.github/workflows/terraform_provider.yml` file.

Version updates:

* [`.devcontainer/devcontainer.json`](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686L6-R6): Updated the base image from `mcr.microsoft.com/devcontainers/base:ubuntu` to `mcr.microsoft.com/devcontainers/base:ubuntu22.04`.
* [`.devcontainer/devcontainer.json`](diffhunk://#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686L21-R21): Updated the Go version from `latest` to `1.22.3`.
* [`go.mod`](diffhunk://#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L3-R3): Updated the Go version from `1.21` to `1.22.3`.

Workflow changes:

* [`.github/workflows/terraform_provider.yml`](diffhunk://#diff-9fa590c928d7324e2af24ae397276bc6a73b89ef6082dfb0eb440b17f527dc0fL73-R103): Commented out the `linter` job.This pull request primarily focuses on improving the GitHub Actions workflow and updating the Go version in the `terraform-provider-power-platform` project. The changes include a more detailed setup for the Go environment in the GitHub Actions workflow and an update to the Go version in the `go.mod` file.

GitHub Actions Workflow Improvements:

* [`.github/workflows/terraform_provider.yml`](diffhunk://#diff-9fa590c928d7324e2af24ae397276bc6a73b89ef6082dfb0eb440b17f527dc0fL84-R105): The actions workflow has been updated to include more detailed steps for setting up Go and running the `tfproviderlint` tool. This includes checking out the code into the Go module directory, setting up Go with caching enabled and the Go version specified in `go.mod`, displaying the Go version and the contents of the directory, installing the Provider Linter, and running `tfproviderlint` on the entire directory. The tfproviderlint will be disabled / commented until it works.

Go Version Update:

* [`go.mod`](diffhunk://#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6L3-R3): The Go version used in the project has been updated from `1.21` to `1.21.9`.This pull request includes a minor update to the Go version in the `go.mod` file. The Go version has been updated from `1.21` to `1.21.9`, which could include bug fixes and minor improvements.

### CODE REVIEW

**Comment by eduardodfmex** [comment]
We change the GO version but also the 

Install the lint: 

Run go install github.com/bflad/tfproviderlint/cmd/tfproviderlint@latest
go: downloading github.com/bflad/tfproviderlint v0.29.0
go: downloading golang.org/x/tools v0.8.0
go: downloading github.com/bflad/gopaniccheck v0.[1](https://github.com/microsoft/terraform-provider-power-platform/actions/runs/8993981823/job/24706773420?pr=284#step:8:1).0
go: downloading golang.org/x/sys v0.7.0
go: downloading golang.org/x/mod v0.10.0


Run the lint:

Run tfproviderlint ./...
  tfproviderlint ./...
  shell: /usr/bin/bash -e {0}

**Review by mawasile — DISMISSED** [review]

**Review by mattdot — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/microsoft/terraform-provider-power-platform
 
-go 1.21
+go 1.21.9
```
Comment (polatengin) [inline go.mod:3]: why don't we move to 1.22 directly? newest release to this day is `1.22.3`

https://go.dev/doc/devel/release

File: .github/workflows/terraform_provider.yml [hunk 2]
```diff
@@ -81,10 +81,28 @@ jobs:
...
       with:
-        args: ./...
+        cache: true
+        go-version-file: 'go.mod'
+      id: go
+    - run: |
+        go version
+
+    - run: |
+        ls -al
```
Comment (polatengin) [inline .github/workflows/terraform_provider.yml:97]: do we need to have the list of files and folder of the working directory in the logs?

File: .github/workflows/terraform_provider.yml [hunk 3]
```diff
@@ -81,10 +81,28 @@ jobs:
...
+
+    - name: Set up Go
+      uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
       with:
-        args: ./...
+        cache: true
+        go-version-file: 'go.mod'
+      id: go
+    - run: |
+        go version
```
Comment (polatengin) [inline .github/workflows/terraform_provider.yml:94]: do we need this, or do we have it debugging purpose?

File: .github/workflows/terraform_provider.yml [hunk 4]
```diff
@@ -81,10 +81,28 @@ jobs:
...
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b # v4.1.5
+
+    - name: Set up Go
+      uses: actions/setup-go@cdcb36043654635271a94b9a6d1392de5bb323a7 # v5.0.1
       with:
-        args: ./...
+        cache: true
+        go-version-file: 'go.mod'
+      id: go
```
Comment (polatengin) [inline .github/workflows/terraform_provider.yml:92]: a white-space could make it easier to understand which task have the `id: go` field

**Review by polatengin — APPROVED** [review]
I have some nit-picky comments

File: .github/workflows/terraform_provider.yml (hunk 3 above)
Comment (eduardodfmex) [reply .github/workflows/terraform_provider.yml:94]: Its more for debugging.!

File: .github/workflows/terraform_provider.yml (hunk 2 above)
Comment (eduardodfmex) [reply .github/workflows/terraform_provider.yml:97]: Thank you for the suggestion.!

Comment (eduardodfmex) [reply .github/workflows/terraform_provider.yml:97]: Is not needed. thks.!

File: go.mod (hunk 1 above)
Comment (eduardodfmex) [reply go.mod:3]: Meke sense, I will update the version.

File: .devcontainer/devcontainer.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
 			"terragrunt": "latest"
 		},
 		"ghcr.io/devcontainers/features/go:1": {
-			"version": "latest"
+			"version": "1.22.2"
```
Comment (polatengin) [inline .devcontainer/devcontainer.json:21]: it'd be better if we use the same version as we have in the `go.mod` file (`1.22.3`)

**Review by polatengin — COMMENTED** [review]
one more nit-picky comment :)

File: .devcontainer/devcontainer.json (hunk 5 above)
Comment (eduardodfmex) [reply .devcontainer/devcontainer.json:21]: Done.

File: .github/workflows/terraform_provider.yml [hunk 6]
```diff
@@ -70,21 +70,37 @@ jobs:
...
+#    - run: |
+#        go version
+
+#    - name: Install Provider Linter
+#      run: |
+#        go install github.com/bflad/tfproviderlint/cmd/tfproviderlint@latest
+
+#    - name: Run tfproviderlint
+#      run: |
+#        tfproviderlint ./...
```
Comment (polatengin) [inline .github/workflows/terraform_provider.yml:103]: is there a reason to disable these steps? if so, I would remove them from the workflow..

**Review by polatengin — COMMENTED** [review]
I left a comment for the disabled code block in the workflow

File: .github/workflows/terraform_provider.yml (hunk 6 above)
Comment (eduardodfmex) [reply .github/workflows/terraform_provider.yml:103]: Thank you for asking, the lint is not working, is failing, we are having this issue related to go 1.22: https://github.com/bflad/tfproviderlint/issues/298

tfproviderlint -R014=false -AT001.ignored-filename-suffixes=_data_source_test.go ./...
  shell: /usr/bin/bash -e {0}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference.......... 

We agree to disable and see if the issue is solved, and the lint will work.

Do you prefer to remove this lint part, we also have other parts in this WF commented.

Comment (eduardodfmex) [reply .github/workflows/terraform_provider.yml:103]: As we discuse we will keep commented, once we implement a second lint we can remove this part.

**Review by polatengin — APPROVED** [review]

**Review by mattdot — APPROVED** [review]

**Pull request merged**